tasks/prepare-binaries.js

```diff
@@ -71,28 +71,38 @@ function copyAssets (platforms) {
 
 async function archive (platforms) {
   console.log('Zipping...')
-  await Promise.all(Object.keys(platforms).map(async (platform) => {
+
+  const promises = Object.keys(platforms).map(async (platform) => {
+    const rootname = `${appName}-${platform}`
     const archive = archiver('zip', {
       zlib: { level: 9 } // Maximize compression
     })
 
-    // must not be in same dir where we are zipping
-    const zipOut = fs.createWriteStream(path.join(buildDirPath, `${appName}-${platform}.zip`))
-    zipOut.on('close', function () {
-      console.log(`Wrote ${Math.round(archive.pointer() / 1e4) / 1e2} Mb total to ${platform}`)
-    })
-    archive.on('error', function (err) {
-      throw err
-    })
+    const zipOut = fs.createWriteStream(path.join(buildDirPath, `${rootname}.zip`))
+
     // pipe archive to file stream
     archive.pipe(zipOut)
+
+    // must not be in same dir where we are zipping
+    const p0 = new Promise((resolve, reject) => {
+      zipOut.on('close', function () {
+        console.log(`Wrote ${Math.round(archive.pointer() / 1e4) / 1e2} Mb total to ${platform}`)
+        resolve()
+      })
+      zipOut.on('error', reject)
+      archive.on('error', reject)
+    })
+
     // recursively add directory to _root_ of zip
-    archive.directory(path.join(buildDirPath, platform), false)
-    await archive.finalize()
-  }))
+    archive.directory(path.join(buildDirPath, platform), rootname, {})
+
+    const p1 = archive.finalize()
+    return Promise.all([p0, p1])
```

Comment:
I think we should use explicit name instead of `p0` and `p1`. What do you think about:

- `archiveFinalize`
- `zipClose`